egui/src/grid.rs

```diff
@@ -178,15 +178,13 @@ impl GridLayout {
             widget_rect.height().at_least(self.min_cell_size.y),
         );
 
+        cursor.min.x += self.prev_col_width(self.col) + self.spacing.x;
         self.col += 1;
-        cursor.min.x += frame_rect.width() + self.spacing.x;
     }
 
     pub(crate) fn end_row(&mut self, cursor: &mut Rect, painter: &Painter) {
-        let row_height = self.prev_row_height(self.row);
-
         cursor.min.x = self.initial_x;
-        cursor.min.y += row_height + self.spacing.y;
+        cursor.min.y += self.curr_state.row_height(self.row).unwrap_or(self.min_cell_size.y) + self.spacing.y;
```

Comment:
This makes sense to me - we know how high the row turned out to be, so we should use that instead of the height from previous frame. 👍